apps/src/netsim/NetSimVizWire.js

```diff
@@ -192,29 +174,33 @@ NetSimVizWire.prototype.setEncodings = function(newEncodings) {
  * Kick off an animation of the wire state being set by the local viznode.
  * @param {"0"|"1"} newState
  */
-NetSimVizWire.prototype.animateSetState = function(newState) {
+NetSimVizWire.prototype.animateSetState = function(newState, fromRemote) {
   if (!(this.localVizNode && this.remoteVizNode)) {
     return;
   }
 
   var flyOutMs = 300;
-  var holdPositionMs = 300;
 
   this.stopAllAnimation();
   this.setWireClasses_(newState);
   this.text_.text(this.getDisplayBit_(newState));
-  this.snapTextToPosition(this.getLocalNodePosition());
+  this.snapTextToPosition(
+    fromRemote ? this.getRemoteNodePosition() : this.getLocalNodePosition()
+  );
   this.tweenTextToPosition(
     this.getWireCenterPosition(),
     flyOutMs,
     tweens.easeOutQuad
   );
-  this.doAfterDelay(
-    flyOutMs + holdPositionMs,
-    function() {
-      this.setWireClasses_('unknown');
-    }.bind(this)
-  );
+};
+
+NetSimVizWire.prototype.changeWireState = function(newState) {
```

Comment:
I've got a mild naming preference for `setWireState` here, since you could call this with the existing state and reasonably expect idempotency.